### PR TITLE
Fix mypy apache beam operators

### PR DESCRIPTION
**Pull request by subkanthi**
related: #19891 
Fixes the following mypy errors.
```
airflow/providers/apache/beam/operators/beam.py:255: error: Item "None" of "Optional[DataflowHook]" has no attribute "provide_authorized_gcloud"
                    with self.dataflow_hook.provide_authorized_gcloud():
                         ^
airflow/providers/apache/beam/operators/beam.py:266: error: Item "None" of "Optional[DataflowHook]" has no attribute "wait_for_done"
                    self.dataflow_hook.wait_for_done(
                    ^
airflow/providers/apache/beam/operators/beam.py:267: error: Argument "job_name" to "wait_for_done" of "DataflowHook" has incompatible type "Optional[str]"; expected
"str"
                        job_name=dataflow_job_name,
                                 ^
airflow/providers/apache/beam/operators/beam.py:268: error: Argument "location" to "wait_for_done" of "DataflowHook" has incompatible type "Optional[str]"; expected
"str"
                        location=self.dataflow_config.location,
                                 ^
airflow/providers/apache/beam/operators/beam.py:432: error: Item "None" of "Optional[DataflowHook]" has no attribute "is_job_dataflow_running"
                            self.dataflow_hook.is_job_dataflow_running(
                            ^
airflow/providers/apache/beam/operators/beam.py:444: error: Item "None" of "Optional[DataflowHook]" has no attribute "is_job_dataflow_running"
                            is_running = self.dataflow_hook.is_job_dataflow_running(
                                         ^
airflow/providers/apache/beam/operators/beam.py:450: error: Item "None" of "Optional[DataflowHook]" has no attribute "provide_authorized_gcloud"
                        with self.dataflow_hook.provide_authorized_gcloud():
                             ^
airflow/providers/apache/beam/operators/beam.py:457: error: Item "None" of "Optional[DataflowHook]" has no attribute "wait_for_done"
                        self.dataflow_hook.wait_for_done(
                        ^
airflow/providers/apache/beam/operators/beam.py:458: error: Argument "job_name" to "wait_for_done" of "DataflowHook" has incompatible type "Optional[str]"; expected
"str"
                            job_name=dataflow_job_name,
                                     ^
airflow/providers/apache/beam/operators/beam.py:459: error: Argument "location" to "wait_for_done" of "DataflowHook" has incompatible type "Optional[str]"; expected
"str"
                            location=self.dataflow_config.location,
                                     ^
airflow/providers/apache/beam/operators/beam.py:461: error: Argument "multiple_jobs" to "wait_for_done" of "DataflowHook" has incompatible type "Optional[bool]";
expected "bool"
                            multiple_jobs=self.dataflow_config.multiple_jobs,
```

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/apache/beam/operators/beam.py [hunk 1]
```diff
@@ -263,12 +263,13 @@ def execute(self, context: 'Context'):
...
-                self.dataflow_hook.wait_for_done(
-                    job_name=dataflow_job_name,
-                    location=self.dataflow_config.location,
-                    job_id=self.dataflow_job_id,
-                    multiple_jobs=False,
-                )
+                if dataflow_job_name and self.dataflow_config.location:
+                    self.dataflow_hook.wait_for_done(
+                        job_name=dataflow_job_name,
+                        location=self.dataflow_config.location,
```
Comment (subkanthi) [inline airflow/providers/apache/beam/operators/beam.py:269]: assumption is the job_name and self.dataflow_config.location are required parameters

File: airflow/providers/apache/beam/operators/beam.py [hunk 2]
```diff
@@ -454,13 +455,22 @@ def execute(self, context: 'Context'):
...
-                        multiple_jobs=self.dataflow_config.multiple_jobs,
-                        project_id=self.dataflow_config.project_id,
-                    )
+                    if dataflow_job_name and self.dataflow_config.location:
+                        if self.dataflow_config.multiple_jobs:
+                            self.dataflow_hook.wait_for_done(
+                                job_name=dataflow_job_name,
+                                location=self.dataflow_config.location,
+                                job_id=self.dataflow_job_id,
+                                multiple_jobs=self.dataflow_config.multiple_jobs,
```
Comment (subkanthi) [inline airflow/providers/apache/beam/operators/beam.py]: maybe can be re-written better.

Comment (potiuk) [reply airflow/providers/apache/beam/operators/beam.py]: I think I already fixed that one in #20611 -> rebase after I merge and likely this one will not be needed.

Comment (potiuk) [reply airflow/providers/apache/beam/operators/beam.py]: You might try to rebase now :)

Comment (subkanthi) [reply airflow/providers/apache/beam/operators/beam.py]: looks like its still needed.

```
airflow/providers/apache/beam/operators/beam.py:463: error: Argument "multiple_jobs" to "wait_for_done" of "DataflowHook" has incompatible type "Optional[bool]";
expected "bool"
                                multiple_jobs=self.dataflow_config.multiple_jobs,
```

Comment (potiuk) [reply airflow/providers/apache/beam/operators/beam.py]: Ah right. That was another "wait".

Comment (potiuk) [reply airflow/providers/apache/beam/operators/beam.py]: Easy - default is False if not set.

```
mutltiple_jobs = self.dataflow_config.mutliple_jobs if self.dataflow_config.multiple_jobs else False

....
 
        mutltiple_jobs=multiple_jobs,
```

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Some tests failing though :(

**Comment by subkanthi** [comment]
> Some tests failing though :(

Fixed now.

**Review by potiuk — APPROVED** [review]

**Pull request merged**